### ISSUE TITLE
[YSQL] Restarting Postmaster after crash while holding LWLock results in stats collector segfaults

### ISSUE THREAD

**Issue by timothy-e** [yugabyte/yugabyte-db#19572]
Jira Link: [DB-8352](https://yugabyte.atlassian.net/browse/DB-8352)
### Description

Run
```sh
bin/yb-ctl restart --tserver_flags=TEST_yb_lwlock_crash_after_acquire_pg_stat_statements_reset=true
bin/ysqlsh -c "SELECT pg_stat_statements_reset();"
```

Observe in the logs:
```
2023-10-17 14:20:00.008 EDT [83152] LOG:  statistics collector process (PID 83240) was terminated by signal 11: Segmentation fault
```

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-8352]: https://yugabyte.atlassian.net/browse/DB-8352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by timothy-e** [comment]
Is not present on commit 70fbb382e1a913c651a6dbc85a75bcf2e25d32a8 (HEAD)
Author: Tim Elgersma <timothy.elgersma@gmail.com>
Date:   Mon Dec 12 16:41:08 2022 -0500

    [#14614] YSQL: Cleanup after terminated connections

**Comment by timothy-e** [comment]
Closed by 466d5108b43c030d4e2b7096500153fcdaa238c8
